bin/install.sh

```diff
@@ -5,6 +5,7 @@ declare root=""
 root="$(cd "$(dirname "$(dirname "${BASH_SOURCE[0]}")")" && pwd)"
 
 source "$root/bin/functions.sh"
+declare host="$(host_os)"
```

Comment:
Why does the location of a local variable need to be before a sourced script? Seems like it should be an exported environment variable

Reply:
Assuming you meant 'after', not before.

Because it's used throughout the script.

Reply:
host_os() can translate an argument, or return the translated host for `uname -s` if no argument is provided.
This will be useful in the future if we want to implement cross compile.

Reply:
Is there a reason to avoid calling `host_os` multiple times? Why not define a `host` variable scoped to the file?

Reply:
> Assuming you meant 'after', not before.
> 
> Because it's used throughout the script.

I meant before `source "$root/bin/android-functions.sh"`

Reply:
> Why does the location of a local variable need to be before a sourced script? Seems like it should be an exported environment variable

I really don't understand.

It's not a location and it's not a local variable. 
`host_os` is a function.
Some of the functions support passing in $host, otherwise they default to `$(host_os)`.
This is for testing and general flexibility. 
globals should be avoided where possible.

Reply:
> Is there a reason to avoid calling `host_os` multiple times? Why not define a `host` variable scoped to the file?

Using globals as bug prone and less flexible.

Reply:
I agree. I am just confused why this declaration is on line 8 between to `source` calls, removed from line 31

Reply:
`android-functions.sh` did rely on $host being set globally. It is fixed by calling `$(host_os)` now:
https://github.com/socketsupply/socket/blob/android-fte/bin/android-functions.sh#L407

Yes, there's some globals there. That will take some work to fix.

Reply:
@jwerle I have reverted the change and resolved the reason it was originally required.